Style edits (#122)
ROCm-Developer-Tools/ROC_SHMEM: docs/install.rst

.. meta::
  :description: Instruction on how to install rocSHMEM.
  :keywords: rocSHMEM, ROCm, install, build, dependencies, MPI, UCX, Open MPI

.. _install-rocshmem:

---------------------------
Installing rocSHMEM
---------------------------

> This topic describes how to install rocSHMEM. For more information, see `README.md <https://github.com/ROCm/rocSHMEM/blob/rocm-6.4.0/README.md>`_ in the rocSHMEM GitHub repository.

Reviewer: Why does the user need to go to the Readme - why can't that info go here?

Author: Removed in https://github.com/ROCm/rocSHMEM/pull/122/commits/995d67aa1b61a8101b46b5f72c12e77308093e1f

> Requirements

Reviewer: This list shouldn't be numbered. It should be bulleted.

Author: Fixed in https://github.com/ROCm/rocSHMEM/pull/122/commits/995d67aa1b61a8101b46b5f72c12e77308093e1f

---------------------------

> 1. ROCm v6.4.0 or later, including the :doc:`HIP runtime <hip:index>`.

Reviewer: ```suggestion
1. ROCm 6.4.0 or later, including the :doc:`HIP runtime <hip:index>`.
```

2. AMD GPUs

   * MI250X

   * MI300X

3. ROCm-aware Open MPI and UCX. For more information, see :ref:`install-dependencies`.

> Installing from a Package Manager

Reviewer: ```suggestion
Installing from a package manager
```

Reviewer: Sentence case. Or is package manager a proper name?

---------------------------------

On Ubuntu, you can install rocSHMEM by running:

.. code-block:: bash

   apt install rocshmem-dev

.. note::

>   This installation method requires ROCm v6.4 or later. You must build dependencies such as Open MPI and UCX manually, because the distribution packaged versions do not include full accelerator support. For more information, see :ref:`install-dependencies`.

Reviewer: ```suggestion
  This installation method requires ROCm 6.4 or later. You must manually build dependencies such as Open MPI and UCX, because the distribution packaged versions don't include full accelerator support. For more information, see :ref:`install-dependencies`.
```

Reviewer: Don't use "v" with ROCm version numbers.

.. _install-dependencies:

Building Dependencies
---------------------------

> rocSHMEM requires ROCm-Aware Open MPI and UCX. Other MPI implementations, such as MPICH, are not fully tested.

Reviewer: ```suggestion
rocSHMEM requires ROCm-Aware Open MPI and UCX. Other MPI implementations, such as MPICH, have not been fully tested.
```

> To build and configure ROCm-Aware UCX v1.17.0 or later, run:

Reviewer: probably remove the v unless UCX uses the v

Reviewer 2: UCX uses the v

Author: Fixed in https://github.com/ROCm/rocSHMEM/pull/122/commits/995d67aa1b61a8101b46b5f72c12e77308093e1f

  make -j 8
  make -j 8 install

Alternatively, you can use a script to install dependencies:

.. code-block:: bash

  export BUILD_DIR=/path/to/not_rocshmem_src_or_build/dependencies
  /path/to/rocshmem_src/scripts/install_dependencies.sh

.. note::

  Configuration options vary by platform. Review the script to ensure it is compatible with your system.

For more information about OpenMPI-UCX support, see
`GPU-enabled Message Passing Interface <https://rocm.docs.amd.com/en/latest/how-to/gpu-enabled-mpi.html>`_.

> Installing rocSHMEM from Source

Reviewer: ```suggestion
Installing rocSHMEM from source
```
